.github/workflows/helm-lint-test.yaml

```diff
@@ -0,0 +1,161 @@
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# #############################################################################
+---
+name: Helm Checks (Lint, Test, Kyverno policies)
+on:
+  pull_request:
+    paths:
+      - 'charts/**'
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      node_image:
+        description: 'kindest/node image for k8s kind cluster'
+        # k8s version from 3.1 release as default
+        default: 'kindest/node:v1.27.3'
+        required: false
+        type: string
+      ## Skip upgrade for now until a working chart is released
+      #upgrade_from:
+      #  description: 'chart version to upgrade from'
+      #  # chart version from 3.1 release as default
+      #  default: 'x.x.x'
+      #  required: false
+      #  type: string
+      helm_version:
+        description: 'helm version to test (default = latest)'
+        default: 'latest'
+        required: false
+        type: string
+
+jobs:
+  list-changed:
+    runs-on: ubuntu-latest
+    outputs:
+      changed: "${{ steps.list-changed.outputs.changed }}"
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Helm
+      uses: azure/setup-helm@v3
+      with:
+        version: ${{ github.event.inputs.helm_version || 'latest' }}
+        token: ${{ secrets.GITHUB_TOKEN }}
+
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.10'
+        check-latest: true
+
+    - name: Set up chart-testing
+      uses: helm/chart-testing-action@v2.6.1
+
+    - name: Run chart-testing (list-changed)
+      id: list-changed
+      run: |
+        changed=$(ct list-changed --target-branch ${{ github.event.repository.default_branch }})
+        if [[ -n "$changed" ]]; then
+          echo "changed=true" >> $GITHUB_OUTPUT
+        fi
+
+  lint-test:
+    runs-on: ubuntu-latest
+    needs: [list-changed]
+    if: ${{ github.event_name != 'pull_request' || needs.list-changed.outputs.changed == 'true' }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Helm
+      uses: azure/setup-helm@v3
+      with:
+        version: ${{ github.event.inputs.helm_version || 'latest' }}
+        token: ${{ secrets.GITHUB_TOKEN }}
+
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.10'
+        check-latest: true
+
+    - name: Kubernetes KinD Cluster
+      uses: container-tools/kind-action@v2
+      with:
+        # upgrade version, default (v0.17.0) uses node image v1.21.1 and doesn't work with more recent node image versions
+        version: v0.20.0
+        # default value for event_name != workflow_dispatch
+        node_image: ${{ github.event.inputs.node_image || 'kindest/node:v1.27.3' }}
+
+    - name: Set up chart-testing
+      uses: helm/chart-testing-action@v2.6.1
```

Comment:
Yes, python is a requirement of chart-testing. KinD is an interesting one. I was following the example from this [TRG](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-09#description) but it might only be used for the upgradeability part that we don't have it here yet. I will test the job without this test. As for the kind-action source, the one we use now [reimplements the helm/kind-action](https://github.com/container-tools/kind-action?tab=readme-ov-file#credits) so I think we can use either of them.

Reply:
In [this example](https://github.com/helm/chart-testing-action?tab=readme-ov-file#example-workflow) they are using the kind-action to spin up a cluster and run the chart testing there. I've adjusted the workflow similarly to this one.